Implemented empty filter constraint, fixed crash on empty queries

ilyahryapko opened the pull request:
If users specifies empty mongo filter BaseRepository crashes in BuildFilterQuery return statement:
`"$and/$or/$nor must be a nonempty array"`
Fixed that issue.

Also implemented (copypasted Stepan's solution from ServGrow) constraint on empty filter as caution for developer when filter is used in update operations

goremukin commented on src/app/Common/DAL/BaseFilter.cs:
```diff
@@ -3,5 +3,6 @@
     public class BaseFilter
     {
         public string Id { get; set; }
+        public bool EmptyFilterAllowed { get; set; }
```
I prefer AllowEmptyFilter or IsEmptyFilterAllowed. But it is not critical

ilyahryapko replied:
> Looks good. Did you test existing functionality?

I've ran into that problem on Servgrow. He're same logic, so yeah.
Also implemented "FindAsync" method for testing purposes in this solution (removed before comitting): error reproduced. That's why I've made this little fix. Also implemented AllowEmptyFilter for the reasons I've mentioned in the PR message.

As for the name: I've named it using my personal taste, It's not critical to rename it. You can check MSDN naming convention: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members. TLDR: `"Optionally, you can also prefix Boolean properties with "Is", "Can", or "Has", but only where it adds value"`. I think `V + ed => participle` describes this property being a boolean. As far as I know you, we will not agree with points of each other. So I think we need Sergey's opinion aswell

goremukin reviewed (commented):
Looks good. Did you test existing functionality?

ilyahryapko commented:
> Looks good. Did you test existing functionality?

I've ran into that problem on Servgrow. Same logic is used here, so yeah.
Also implemented "FindAsync" method for testing purposes in this solution (removed before comitting): error reproduced. That's why I've made this little fix. Also implemented AllowEmptyFilter for the reasons I've mentioned in the PR message.

As for the name: I've named it using my personal taste, It's not critical to rename it. You can check MSDN naming convention: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members. TLDR: `"Optionally, you can also prefix Boolean properties with "Is", "Can", or "Has", but only where it adds value"`. I think `V + ed => participle` describes this property being a boolean. As far as I know you, we will not agree with points of each other. So I think we need Sergey's opinion aswell

goremukin commented:
@ilyahryapko, Just in case make sure that empty filters are not used anywhere, these are different projects a little.

About the naming. I constantly see Can, Has, and Is prefixes in the standard Microsoft libraries and in our projects in Paralect. Besides, I think it really reads better and with this prefix it is easier to find what you need. @SergeyLukashov,  What do you think?

ilyahryapko commented:
> @ilyahryapko, Just in case make sure that empty filters are not used anywhere, these are different projects a little.
Yep, checked. There's only two inheritors of BaseFilter and both are supposed to be not empty

SergeyLukashov reviewed (approved):
Looks good. And all the variable names are fine 🙂

goremukin commented:
It's terribly inconvenient to re-view changes after changing line endings. But let's merge
